### PR TITLE
Integrate with AtomsBase 0.3

### LINKED ISSUE #26: Issues with AtomsBase interface
When integrating ExtXYZ with [AtomsIO](https://github.com/mfherbst/AtomsIO.jl) I found a bunch of issues in the AtomsBase interface, which currently lead me to rewrite most of it in AtomsIO, see https://github.com/mfherbst/AtomsIO.jl/blob/master/src/extxyz.jl. Of course better this should be integrated here.

One concern for me at the moment are the lines
https://github.com/libAtoms/ExtXYZ.jl/blob/2c35dcb51728dd2e458807332a2c4fe27c31c8bb/src/atoms.jl#L53--L66
which effectively strip off all units of all data stored in the `ExtXYZ.Atoms`. I think this is problematic, because it is later impossible to recover the unit associated to a key. I think it should be the responsibility of the user to do the intrusive stripping off of the units and rather just warn about unsupported unitful quantities and ignore them. My rationale would be that it is better to ignore a key rather than conveying information, which could be misleading accidentality. This is the approach I have followed in AtomsIO. Essentially I only support unitless quantities or unitful quantities, which follow a convention, which is documented in AtomsBase (see [this PR](https://github.com/JuliaMolSim/AtomsBase.jl/blob/300cfe31da1b57f51e5297f149d476627192b7ab/docs/src/tutorial.md)).

@jameskermode What is your opinion about this?

### PR DESCRIPTION
**Pull request by mfherbst**
Closes #26, additionally:

- Integrates with arbitrary property interface from AtomsBase 0.3
- Restructures tests
- Updates CI workflow
- Updated the readme
- Bumps version to 0.1.10

### CODE REVIEW

File: test/atomsbase.jl [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    @test info["charge"]       == ustrip.(u"e_au", sysprop.charge)
+    @test info["extra_data"]   == sysprop.extra_data
+    @test info["multiplicity"] == sysprop.multiplicity
+
+    arrays = atoms["arrays"]
+    @test arrays["Z"]          == atprop.atomic_number
+    @test arrays["species"]    == string.(atprop.atomic_symbol)
+    @test arrays["mass"]       == ustrip.(u"u",   atprop.atomic_mass)
+    @test arrays["pos"]        ≈  ustrip.(u"Å",   hcat(atprop.position...)) atol=1e-10
+    @test arrays["velocities"] ≈  ustrip.(u"Å/s", hcat(atprop.velocity...)) atol=1e-10
```
Comment (jameskermode) [inline test/atomsbase.jl:35]: As noted elsewhere this unit is wrong

**Review by jameskermode — COMMENTED** [review]
This looks good, thank you!

My one reservation is that I don't like the use of hard-coded lists of properties and the consequent fixed mapping of particular string names to units. This is OK for the truly mandatory ExtXYZ properties (basically just lattice, positions, chemical species) and by extension to the other 'standard' AtomsBase properties but I'm not sure how to deal with anything beyond this. 

I suppose just leaving them unitless as you have done is the only we can do.

**Comment by mfherbst** [comment]
If you confirm `u"eV^0.5/u^0.5"` in the other PR, then I'll use that here as well.

> My one reservation is that I don't like the use of hard-coded lists of properties

yeah this is tricky. My reasoning is that: If there is a preferred name that has a well-adjusted meaning, then it ought to have a well-defined unit as well. So you are bound to check this explicitly. In the long run (when the list gets longer) we could provide proper constants in `AtomsBase` that essentially export both the system / atom keywords along with the associated unit. Then one does not have to hard-code this in here (and everywhere else).

**Comment by mfherbst** [comment]
@jameskermode I changed the units as discussed in the other PR. Let me know if you have other things to be done before merging this.

**Comment by jameskermode** [comment]
Happy to merge now. Thanks for the contribution!

**Comment by jameskermode** [comment]
Release 0.1.10 tagged and will show up in General registry shortly.

**Comment by mfherbst** [comment]
Thanks!

**Pull request merged**